dpcpp/matrix/sparsity_csr_kernels.dp.cpp

```diff
@@ -303,7 +303,23 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
 template <typename ValueType, typename IndexType>
 void sort_by_column_index(std::shared_ptr<const DpcppExecutor> exec,
                           matrix::SparsityCsr<ValueType, IndexType>* to_sort)
-    GKO_NOT_IMPLEMENTED;
+{
+    const auto num_rows = to_sort->get_size()[0];
+    const auto row_ptrs = to_sort->get_const_row_ptrs();
+    const auto cols = to_sort->get_col_idxs();
+    auto queue = exec->get_queue();
+    // build sorted postorder node list for each row
+    queue->submit([&](sycl::handler& cgh) {
+        cgh.parallel_for(sycl::range<1>{num_rows}, [=](sycl::id<1> idx_id) {
+            const auto row = idx_id[0];
+            const auto row_begin = row_ptrs[row];
+            const auto row_end = row_ptrs[row + 1];
+            // heap-sort the elements
+            std::make_heap(cols + row_begin, cols + row_end);
+            std::sort_heap(cols + row_begin, cols + row_end);
```

Comment:
I guess the reason you use heap sort here is that std::sort is not available in kernels?

Reply:
heapsort doesn't need recursion (at least the libstdc++ version is unrolled), while std::sort may use recursive calls.